roles/network_plugin/weave/tasks/pre-upgrade.yml

```diff
@@ -37,3 +37,5 @@
 
 - name: Weave pre-upgrade | Purge legacy weave docker containers
   shell: "docker ps -af 'name=^/weave.*' -q | xargs --no-run-if-empty docker rm -f"
+  retries: 3
```

Comment:
docker container removal fails on centos a lot because the devicemapper driver is.... amazing. let's leave it in because it seems to help